File: leetcode-submissions/1760_Minimum_Limit_of_Balls_in_a_Bag.cpp
class Solution {
public:
    bool isPossible(int mid,vector<int>&nums,int maxop){
        int totalOperations=0;

        for(int num:nums){
            int op=num/mid;
            if(num%mid==0){
                op-=1;
            }
            totalOperations+=op;
            
        }
        if(totalOperations>maxop){
            return false;
        }
        return true ;
    }
    int minimumSize(vector<int>& nums, int maxOperations) {
        int right =INT_MIN ;
        for(int n:nums){
            right=max(right,n);
        }
        int left=1;
        int result=0;

        while(left<=right){
            int mid=(right+left)/2;
            if(isPossible(mid,nums,maxOperations)){
result=mid;
right=mid-1;

            }else{
                left=mid+1;
            }
        }
        return result ;
    }
};

File: leetcode-submissions/1760_Minimum_Limit_of_Balls_in_a_Bag.ts
function minimumSize(nums: number[], maxOperations: number): number {
    let maxEle: number = Math.max(...nums); // Find the maximum element in the array
    let left: number = 1, right: number = maxEle;
    let result: number = maxEle;

    while (left <= right) {
        let mid: number = Math.floor((left + right) / 2);

        if (isPossible(mid, nums, maxOperations)) {
            result = mid; // If mid is possible, it could be the minimum valid size
            right = mid - 1; // Try for a smaller size
        } else {
            left = mid + 1; // Increase size if mid is not possible
        }
    }
    return result;
}

function isPossible(mid: number, nums: number[], maxOperations: number): boolean {
    let totalOperations: number = 0;
    for (const num of nums) {
        let op: number = Math.ceil(num / mid) - 1; // Number of operations needed for this element
        totalOperations += op; // Accumulate total operations needed
    }
    return totalOperations <= maxOperations;
}
